### DISCUSSION TITLE
Problem FPA golf 7.5 GTE

### DISCUSSION THREAD

**Discussion by Libar19** [jilleb/MQB-FPA#87] [Q&A]
Hello everyone. I have a golf 7.5 GTE and I wanted to add driving modes to my GTE. I replaced my gateway with a 3Q0 907 530 AD as I was advised. I removed the component protection and the car was fully functional (GTE and e-mode). after having injected the dataset (transfer completed successfully), I no longer have any driving mode.. so I tried many other datasets but none worked to reactivate my GTE and e-mode modes (and I don't is not the co-driving mode either)... the FECs are all good so I don't understand where the problem could come from, I left the car at rest but the same...

**Comment by Carmaffia** [comment]
I think the dataset is specific for the pre-facelift Golf 7. You need to look for the OEM dataset, or edit one to suit your needs. There's a online FPA editor available:  http://www.mqbtools.nl/fpa/

**Reply by razvanphp** [reply]
Any hint on how to chose the base dataset to start from?

I'm in the same boat, but worse: 2015 GTE preFL, tried to swap CAN gateway 5QE907530C with 3Q0907530AD, but ODIS online removed CP and stopped at parametrisation with error because my car never had this part number module.

I guess we can try the hack with removing ECU and Instruments fuse and put a manual VIN in ODIS online from a car with driving modes from factory. Would that work or is there a better approach?

We also tried VCP with [those datasets](https://github.com/chobott/MQB-19-Gateway-EV-GTE) files, but can't manage to write them... VCP wants .zdc files; can I just convert them?

**Reply by Carmaffia** [reply]
_I'm in the same boat, but worse: 2015 GTE preFL, tried to swap CAN gateway 5QE907530C with 3Q0907530AD, but ODIS online removed CP and stopped at parametrisation with error because my car never had this part number _module.__

I did the same conversion, in the same car without a problem. So give some more details.

**Reply by razvanphp** [reply]
Thank you for the quick answer. Not sure what other details I can give, it was just failing on ODIS online, the parametrisation worked only with the 530C original gateway, even this one flashed with 530E was throwing error, it was saying to check the control module part number or replace it (CAN gateway, that is). Indeed for my VIN in ETKA the latest gateway code is 530C, no replacement, so thought this is the reason.

What did you have on your car? And what year is it?

**Reply by Carmaffia** [reply]
2015 GTE Original Mib1. 5qe907530b. Trick was (i think) to NOT do the replacement route in Odis. But just remove old gateway, plug in AD gateway and then follow the Guided Fault Finding option in Odis. That way, you still have the old CP free gateway as backup. Parametrisation was no problem. There was also a small ECU update needed.  Dataset was needed to unlock all driving modes.

**Reply by razvanphp** [reply]
CP worked fine for me too on both gateways, I can swap them freely now and both work, parametrisation does not work tho...

BTW, where did you find the ECU flash file and what version is it?

**Reply by Carmaffia** [reply]
When i removed CP on gateway i got a error from Odis in ECU. But if you are in GFF i did remove/repair all errors. Do you have a error in AD gateway right now? Maybe you can get away by flashing new firmware on gateway, as i understand firmware also has a universal dataset on it. Can you tell me what firmware version is on your AD gateway right now?
